lang/cs/Org.Apache.REEF.Driver/Task/TaskTransitionState.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Driver.Task
+{
+    /// <summary>
+    /// Task states in task transitions
+    /// </summary>
+    public enum TaskTransitionState
+    {
+        /// <summary>
+        /// When task (task configuration) is first created and added in the task list, its state is TaskNew
+        /// </summary>
+        TaskNew,
+
+        /// <summary>
+        /// After submit a task with an IActiveContext, the task state is changed to TaskSubmitting
+        /// </summary>
+        TaskSubmitting,
+
+        /// <summary>
+        /// After receiving RunningTask event, the task state is TaskRunning
+        /// </summary>
+        TaskRunning,
+
+        /// <summary>
+        /// After receiving CompletedTask event, the task state is TaskCompeleted
+        /// </summary>
+        TaskCompleted,
+
+        /// <summary>
+        /// After driver send command to close a task, the task state is set to TaskWaitingForClose
+        /// </summary>
+        TaskWaitingForClose,
+
+        /// <summary>
+        /// After receiving FaiedTask event and verified it is closed by driver, the state is set to TaskClosedByDriver
+        /// </summary>
+        TaskClosedByDriver,
```

Comment:
Ah, this makes sense. And it gives weight to the suggestion of moving all of this into the IMRU project and making it `internal`, as these task states can only be understood in that context.